src/components/testCaseList/TestCase.tsx

```diff
@@ -49,8 +55,35 @@ const TestCase = (props) => {
             )}
           </IconButton>
         </td>
-        <td>{testCase.title}</td>
-        <td>{testCase.series}</td>
+
+        {displayTitle === testCase.title && <td>{testCase.title}</td>}
+        {displayTitle !== testCase.title && (
+          <td>
+            <Tooltip title={testCase.title} placement="right">
+              <Button
+                data-testid={`test-case-title-${testCase.id}`}
+                name="title"
+              >
+                {displayTitle}
+              </Button>
+            </Tooltip>
+          </td>
+        )}
+
+        {displaySeries === testCase.series && <td>{testCase.series}</td>}
+        {displaySeries !== testCase.series && (
+          <td>
+            <Tooltip title={testCase.series} placement="right">
+              <Button
+                data-testid={`test-case-series-${testCase.id}`}
+                name="series"
+              >
+                {displaySeries}
+              </Button>
+            </Tooltip>
+          </td>
+        )}
+
```

Comment:
```
<td>
   <TruncateText text={testCase.title} maxLength={60} />
</td>
<td>
    <TruncateText text={testCase.series} maxLength={60} />
</td>
```

and then `TruncateText.tsx`

```
import React from "react";
import Button from "@mui/material/Button";
import Tooltip from "@mui/material/Tooltip";
import { truncateInput } from "../../util/Utils";

const TruncateText = ({ text, maxLength = 60 }) => {
  if (text && text.length > maxLength) {
    const displayText = truncateInput(text, maxLength);
    return (
      <Tooltip title={text} placement="right">
        <Button name="title">{displayText}...</Button>
      </Tooltip>
    );
  } else {
    return text;
  }
};

export default TruncateText;

```

Sorry, this increases the work though 😬

Reply:
@adongare Thanks so much for your help Ashok! I really like the idea! I guess I need to think the React way and this is a great example of it! I appreciate it and will try this component way.

Reply:
No worries, we all are learning.

Reply:
@adongare Had some issues because of the change as well as after the merge, but it's finally done. Please have a look again. I really appreciate your help!

Reply:
np, thanks!